server/service/apple_mdm.go

```diff
@@ -822,6 +823,141 @@ func generateEnrollmentProfileMobileconfig(orgName, fleetURL, scepChallenge, top
 	return buf.Bytes(), nil
 }
 
+type mdmAppleCommandRemoveEnrollmentProfileRequest struct {
+	HostID uint `url:"id"`
+}
+
+type mdmAppleCommandRemoveEnrollmentProfileResponse struct {
+	// TODO
+	Err error `json:"error,omitempty"`
+}
+
+func (r mdmAppleCommandRemoveEnrollmentProfileResponse) error() error { return r.Err }
+
+func mdmAppleCommandRemoveEnrollmentProfileEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*mdmAppleCommandRemoveEnrollmentProfileRequest)
+	err := svc.EnqueueMDMAppleCommandRemoveEnrollmentProfile(ctx, req.HostID)
+	if err != nil {
+		return mdmAppleCommandRemoveEnrollmentProfileResponse{Err: err}, nil
+	}
+	return mdmAppleCommandRemoveEnrollmentProfileResponse{}, nil
+}
+
+func (svc *Service) EnqueueMDMAppleCommandRemoveEnrollmentProfile(ctx context.Context, hostID uint) error {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{}, fleet.ActionList); err != nil {
+		return err
+	}
+
+	h, err := svc.ds.HostLite(ctx, hostID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "getting host info for mdm apple remove profile command")
+	}
+
+	// check authorization again based on host info for team-based permissions
+	if err := svc.authz.Authorize(ctx, h, fleet.ActionCommand); err != nil {
+		return err
+	}
+
+	enabled, err := svc.ds.GetNanoMDMEnrollmentStatus(ctx, h.UUID)
+	switch {
+	case err != nil:
+		return ctxerr.Wrap(ctx, err, "getting mdm enrollment status")
+	case !enabled:
+		return fleet.NewUserMessageError(ctxerr.New(ctx, fmt.Sprintf("mdm is not enabled for host %d", hostID)), http.StatusConflict)
+	default:
+		// ok
+	}
+
+	cmdUUID, err := svc.enqueueMDMAppleCommandRemoveEnrollmentProfile(ctx, h.UUID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "enqueuing mdm apple remove profile command")
+	}
+
+	return svc.pollResultMDMAppleCommandRemoveEnrollmentProfile(ctx, cmdUUID, h.UUID)
+}
+
+func (svc *Service) enqueueMDMAppleCommandRemoveEnrollmentProfile(ctx context.Context, hostUUID string) (string, error) {
+	cmd := new(mdm.Command)
+	cmdUUID := uuid.New().String()
+	cmd.CommandUUID = cmdUUID
+	cmd.Command.RequestType = "RemoveProfile"
+	cmd.Raw = []byte(generateMDMAppleCommandRemoveEnrollmentProfile(cmdUUID, apple_mdm.FleetPayloadIdentifier))
+
+	status, res, err := rawCommandEnqueue(ctx, svc.mdmStorage, svc.mdmPushService, cmd, []string{hostUUID}, false, svc.logger)
+	if err != nil {
+		return cmdUUID, ctxerr.Wrap(ctx, err)
+	}
+
+	// Additional logging to aid in debugging unexpected results
+	if status != http.StatusOK {
```

Comment:
Should it return an error if it's not `StatusOK`? Otherwise we may be polling for a result that will never get executed?